### PR TITLE
Fix Django 5.0 compatibility issues

### LINKED ISSUE #9964: Django 5.0 compatibility
### Issue Summary

[Django 4.2 alpha 1 was released last week](https://www.djangoproject.com/weblog/2023/jan/17/django-42-alpha-1-released/), which means that Django's `main` branch is now on 5.0.

Currently, our `main` branch tests are failing against Django's `main` branch. This issue serves as a tracker of any issues with our code that make it incompatible with Django 5.0.

- [`django.utils.timezone.utc`](https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.timezone.utc) has been removed. It is an alias to [`datetime.timezone.utc`](https://docs.python.org/3/library/datetime.html#datetime.timezone.utc) and we should use `datetime.timezone.utc` directly. Might need to check if there's any difference to `pytz.utc`, as that seems to be what Django used in earlier versions.
- `assertFormError` and `assertFormsetError` no longer allows passing response and form context name
  - https://github.com/django/django/pull/15179
- The usage of `QuerySet.iterator()` on a queryset that prefetches related objects without providing the `chunk_size` argument is no longer allowed.
- Support for logging out via GET requests in the `django.contrib.auth.views.LogoutView` and `django.contrib.auth.views.logout_then_login()` is removed.

Blocked by https://github.com/wagtail/django-modelcluster/issues/169.

Will update if I find more.

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #9964.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by gasman** [comment]
@laymonage There may not be another release of django-modelcluster until some time closer to Django 5, so to unblock this I'd suggest adding an `install_extras` line to the Github Actions config to install it from git, as we've previously done here for example: https://github.com/wagtail/wagtail/commit/177bc53aa0407002a05ff588e0a348678700420f

File: wagtail/search/backends/database/fallback.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def _do_search(self):
                 **{self._score_field: Value(None, output_field=models.FloatField())}
             )
 
-        return queryset.iterator()
+        return queryset.iterator(2000)
```
Comment (laymonage) [inline wagtail/search/backends/database/fallback.py:183]: This is the default value that Django uses if we don't pass anything:

https://docs.djangoproject.com/en/4.1/ref/models/querysets/#iterator

> So long as the QuerySet does not prefetch any related objects, providing no value for chunk_size will result in Django using an implicit default of 2000.

> Using `iterator()` on a queryset that prefetches related objects without providing the `chunk_size` is deprecated. In Django 5.0, an exception will be raise.

Not sure if we want to hardcode it here or put it in a variable (or make it configurable).

**Review by laymonage — COMMENTED** [review]
Tests are passing on Postgres, but they don't even run on MySQL due to an issue in the migrations:

```
django.db.utils.OperationalError: (1061, "Duplicate key name 'wagtailcore_workflowstate_page_id_6c962862'")
```

I suspect an index or constraint may have not been cleaned up properly when the field was changed to a generic foreign key. Not sure what changed in Django that could've triggered this, though.

File: wagtail/search/backends/database/fallback.py (hunk 1 above)
Comment (gasman) [reply wagtail/search/backends/database/fallback.py:183]: Happy to hardcode this until a specific request for making it configurable comes up.

File: wagtail/admin/tests/test_account_management.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def test_logout(self):
         self.login()
 
         # Get logout page
-        response = self.client.get(reverse("wagtailadmin_logout"))
+        response = self.client.post(reverse("wagtailadmin_logout"))
```
Comment (gasman) [inline wagtail/admin/tests/test_account_management.py:115]: Do we need to change the behaviour of the logout option in the menu to match? Looks like this is currently just a plain link.

File: client/src/components/Sidebar/menu/ActionMenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    if (e.ctrlKey || e.shiftKey || e.metaKey || (e.button && e.button !== 0)) {
+      return;
+    }
+
+    if (!isDismissed(item, state)) {
+      dispatch({
+        type: 'set-dismissible-state',
+        item,
+      });
+    }
```
Comment (laymonage) [inline client/src/components/Sidebar/menu/ActionMenuItem.tsx:27]: Not sure if we'll ever have an `ActionMenuItem` with a dismissible badge, but I tried it and it works quite well.

File: wagtail/admin/ui/sidebar.py [hunk 4]
```diff
@@ -85,6 +85,47 @@ def __eq__(self, other):
         )
 
 
+@adapter("wagtail.sidebar.ActionMenuItem", base=BaseSidebarAdapter)
+class ActionMenuItem(MenuItem):
```
Comment (laymonage) [inline wagtail/admin/ui/sidebar.py:89]: I haven't added a corresponding class in `wagtail/admin/menu.py`. This telepath component is generic enough to be reusable, it's basically an empty form with a submit button and a customisable `action` and `method`.

Not sure if it's something we'd like to make available now, though.

**Review by gasman — APPROVED** [review]
All looks good to me!

**Pull request merged**